third_party/blink/renderer/platform/heap_observer_set.h

```diff
@@ -14,7 +14,10 @@ namespace blink {
 
 // A set of observers. Ensures list is not mutated while iterating. Observers
 // are not retained.
-template <class ObserverType>
+// [Replay] We made some changes to allow for deterministic hash functions for
+// a subset of all |ObserverType|s.
+template <typename ObserverType,
+          typename ObserverSet = HeapHashSet<WeakMember<ObserverType>>>
```

Comment:
This change (and many related ones in this PR) seems unnecessary if we're able to change the default hash for the observer type to use the record/replay ID, as in https://github.com/replayio/chromium/pull/548 .  Can we try doing that now?  This will make the diff size vs. upstream smaller and keeping the changes cleaner / simpler will pay off over time.